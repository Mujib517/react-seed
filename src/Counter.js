import React, { Component } from "react";


class Counter extends Component {

    constructor() {
        super();

        this.state = { count: 0 }

        this.inc = this.inc.bind(this);
        this.dec = this.dec.bind(this);
    }

    inc() {
        this.setState({ count: this.state.count + 1 });
    }

    dec() {
        this.setState({ count: this.state.count - 1 });
    }

    render() {
        return <>
            <h1>{this.state.count}</h1>
            <button onClick={this.inc}>++</button>
            <button onClick={this.dec}>--</button>
        </>
    }

}


export default Counter;