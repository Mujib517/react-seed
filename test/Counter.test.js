import React from "react";
import Counter from "../src/Counter";
import { shallow, mount } from "enzyme";

describe("Counter Component", () => {

    // it("should match snapshot", () => {
    //     const wrapper = mount(<Counter />);

    //     expect(wrapper).toMatchSnapshot();
    // });

    it("should have initial count value shown", () => {
        const wrapper = mount(<Counter />);

        const h1 = wrapper.find("h1");

        expect(h1.text()).toBe("0");
    });

    it("should have two buttons", () => {
        const wrapper = mount(<Counter />);

        const btn = wrapper.find("button");

        expect(btn).toHaveLength(2);
    });

    it("should  increment the count value when ++ is clicked", () => {
        const wrapper = mount(<Counter />);

        const btn = wrapper.find("button");
        btn.first().simulate("click");

        wrapper.update();

        const h1 = wrapper.find("h1");
        expect(h1.text()).toBe("1");
    });

    it("should  decerment the count value when -- is clicked", () => {
        const wrapper = mount(<Counter />);

        const btn = wrapper.find("button");
        btn.last().simulate("click");

        wrapper.update();

        const h1 = wrapper.find("h1");
        expect(h1.text()).toBe("-1");
    });
});