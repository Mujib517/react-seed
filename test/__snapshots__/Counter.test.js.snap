// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component should match snapshot 1`] = `ReactWrapper {}`;
